src/garage/torch/algos/loss_function_utils.py

```diff
@@ -1,15 +1,61 @@
-"""loss function utilities."""
+"""Differentiable utils typically used when computing loss functions."""
 import torch
 import torch.nn.functional as F
 
 
-def compute_advantages(discount, gae_lambda, max_len, baselines, rewards):
-    """Calculate advantages."""
-    filter = torch.full((1, 1, 1, max_len - 1), discount * gae_lambda)
+def compute_advantages(discount, gae_lambda, max_path_length, baselines,
+                       rewards):
+    """
+    Calculate advantages.
+
+    Calculate advantages using a baseline (value function) according to
+    Generalized Advantage Estimation (GAE)
+    The discounted cumulative sum can be computed using conv2d with filter.
+    filter:
+        [1, (discount * gae_lambda), (discount * gae_lambda) ^ 2, ...]
+        where the length is same with max_path_length.
+
+    Advantages are a discounted cumulative sum.
+
+    if paths is consisted with
+        [ [p_11, p_12, p_13, ... p_1n],
+          [p_21, p_22, p_23, ... p_2n],
+          ...
+          [p_m1, p_m2, p_m3, ... p_mn] ]
+        (n : max_path_length, m: number of paths)
+
+    baselines and rewards are also has same shape.
```

Comment:
these are helpful, but you should also include a summary telling me the meaning of the elements of a numpy-style shape vector.

For instance (please check that this is correct):

```
"""python
Args:
    baselines (torch.Tensor): A 2D vector of value function estimates with
        shape (N, T), where N is the batch dimension (number of episodes) and
        T is the maximum path length experienced by the agent. If an episode
        terminates in fewer than T time steps, the remaining elements in that
        episode should be setto 0.

    rewards (torch.Tensor): A 2D vector of per-step rewards with shape (N, T),
        where N is the batch dimension (number of episodes) and T is the
        maximum path length experienced by the agent. If an episode terminates
        in fewer than T time steps, the remaining elements in that episode
        should be set to 0.
"""
```

Reply:
This is exactly correct explanation of this function! thank you!

Reply:
haha i wrote it, i should know ;)